.github/workflows/pre-commit.yaml

```diff
@@ -33,6 +34,16 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
+      - name: Config Terraform plugin cache
```

Comment:
For each "root" directory (i.e. - terraform default workspace location), our Terraform validate will do a `terraform init` which pulls down all the necessary providers, modules, etc. and that takes a LOOOOOOOOOOOONG time. I haven't used this before but I am hoping by centrally caching the plugin directory, that we can speedup that process and shave off some time